### PR TITLE
Migrate gin to echo with echox and echojwtx

### PR DESCRIPTION
**Pull request by mikemrm**
This migrates to using echo via the x/echox and x/echojwtx libraries.

### CODE REVIEW

File: chart/permissions-api/templates/deployment.yaml [hunk 1]
```diff
@@ -94,11 +94,11 @@ spec:
               protocol: TCP
           livenessProbe:
             httpGet:
-              path: /healthz
+              path: /livez
               port: http
           readinessProbe:
             httpGet:
-              path: /healthz
+              path: /readyz
```
Comment (JAORMX) [inline chart/permissions-api/templates/deployment.yaml:101]: Let's move these to a separate PR.

**Review by JAORMX — CHANGES_REQUESTED** [review]
This mostly LGTM, just have a question and a couple of requests to move some changes to separate PRs.

File: chart/permissions-api/templates/deployment.yaml (hunk 1 above)
Comment (mikemrm) [reply chart/permissions-api/templates/deployment.yaml:101]: This change is part of the echox change. echox provides the readyz endpoint, so I feel this should be part of the same pr.

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -13,9 +12,9 @@ RUN export DEBIAN_FRONTEND=noninteractive \
     && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/helm.gpg] https://baltocdn.com/helm/stable/debian/ all main" | tee /etc/apt/sources.list.d/helm-stable-debian.list \
     && apt-get update \
     && apt-get install --yes --no-install-recommends \
-        bash-completion \
-        helm \
-        uuid-runtime
+    bash-completion \
+    helm \
+    uuid-runtime
```
[the remark this answers is not in the document]
Comment (mikemrm) [reply .devcontainer/Dockerfile:17]: I'll just drop these lines, I don't think it's actually necessary, was automatic formatting.

File: chart/permissions-api/templates/deployment.yaml (hunk 1 above)
Comment (JAORMX) [reply chart/permissions-api/templates/deployment.yaml:101]: ah, I didn't know it was part of that, makes sense. Let's keep it here.

File: .devcontainer/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (JAORMX) [reply .devcontainer/Dockerfile]: it's cool.

File: internal/api/assignments.go [hunk 3]
```diff
@@ -79,17 +73,15 @@ func (r *Router) assignmentsList(c *gin.Context) {
 
 	assignments, err := r.engine.ListAssignments(ctx, role, "")
 	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"message": "error listing assignments", "error": err.Error()})
-		return
+		return echo.NewHTTPError(http.StatusInternalServerError, "error listing assignments").SetInternal(err)
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply internal/api/assignments.go:76]: great!

**Review by JAORMX — APPROVED** [review]
shipit 🚢
